Zend/zend_API.c

```diff
@@ -2483,9 +2483,11 @@ ZEND_API zend_module_entry* zend_register_module_ex(zend_module_entry *module) /
 	}
 
 	name_len = strlen(module->name);
-	lcname = zend_string_alloc(name_len, module->type == MODULE_PERSISTENT);
+	lcname = zend_string_alloc(name_len, module_type == MODULE_PERSISTENT);
 	zend_str_tolower_copy(ZSTR_VAL(lcname), module->name, name_len);
 
+	int module_number = zend_next_free_module();
```

Comment:
Why do you call ``zend_next_free_module();`` at this point?
May be it's better to delay it until the ``module->module_number`` asisgnment?

Reply:
`zend_next_free_module` does `zend_hash_num_elements(&module_registry) + 1;`. So if I move it after adding, the module numbers start at 2.

Reply:
OK. I don't see any problems. I think the old PR may be merged into old branches and this one into master.